src/components/pages/Chat/MessageAggregator.tsx

```diff
@@ -53,18 +55,24 @@ export default function MessageAggregator({ children }: Props) {
 
     const deleteRoom = useDeleteRoom()
 
+    const loadLocalMessages = useLoadLocalMessages()
+
     useEffect(() => {
         const { current: cache } = cacheRef
 
-        if (!roomId) {
-            return
-        }
+        ;(async () => {
+            if (!roomId) {
+                return
+            }
+
+            cache[roomId] = await loadLocalMessages()
```

Comment:
When we change the room this `useEffect` overwrites anything in `cache[roomId]` from the local db. Meaning, there's no real reason to keep all rooms in `cache`. Just the current one (`cache` can be an array).

Reply:
In `onTextMessage` we have a line
```js
cache[streamId].push(data)
```
There's no point in doing that for room other than the current one.